File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<int> food) 
{
    string str1 = "";
    string str2 = "";
    
    for(int i=1; i<food.size(); i++)
    {
        int f = food[i] / 2;
        for(int j=0; j<f; j++)
        {
            str1 += to_string(i);
        }
    }
    for(int i=food.size() - 1; i>=1; i--)
    {
        int f = food[i] / 2;
        for(int j=0; j<f; j++)
        {
            str2 += to_string(i);
        }
    }
    return str1 + "0" + str2;
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.cs
using System;
using System.Text;

public class Solution 
{
    public string solution(int[] food) 
    {
        StringBuilder tmp = new StringBuilder();
        for(int i=1; i<food.Length; i++)
        {
            int count = food[i] / 2;
            for(int j=0; j<count; j++)
            {
                tmp.Append(i.ToString());
            }
        }
        char[] chars = tmp.ToString().ToCharArray();
        Array.Reverse(chars);
        string reversed = new string(chars);
        tmp.Append("0");
        tmp.Append(reversed);
        
        return tmp.ToString();
    }
}